integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/embedding_retriever.py

```diff
@@ -71,6 +75,7 @@ def __init__(
         self._filters = filters or {}
         self._top_k = top_k
         self._num_candidates = num_candidates
+        self._filter_policy = FilterPolicy.from_str(filter_policy) if isinstance(filter_policy, str) else filter_policy
```

Comment:
Looking at edge cases, would it be helpful here to raise a type error if lets say in integer is passed? Else it would break anyways later in the code.  e.g.
raise TypeError("filter_policy must be either a str or an instance of FilterPolicy")

Reply:
Yeah, we don't check for those typing errors, the types are defined in the signature and are assumed to be followed by the user